components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/CanBeSubclassDeterminer.scala

```diff
@@ -77,8 +77,17 @@ trait CanBeSubclassDeterminer {
         case _ => ().validNel
       }
     }
+    val dataValueRestriction = (_: Unit) => {
+      (givenType, superclassCandidate) match {
+        case (TypedObjectWithValue(_, givenValue), TypedObjectWithValue(_, candidateValue)) if givenValue == candidateValue =>
+          ().validNel
+        case (TypedObjectWithValue(_, givenValue), TypedObjectWithValue(_, candidateValue)) =>
+          s"Types with value have different values: $givenValue and $candidateValue".invalidNel
+        case _ => ().validNel
```

Comment:
Am I right that this means that `Typed[String].canBeSubclassOf(TypedObjectWithValue(Typed[String], "foo"))`?

Reply:
After discussion it is ok - please just leave a comment that we can't compute values in all situations so for method arguments we should be lax in values checking or sth similar